src/lightning_app/runners/cloud.py

```diff
@@ -224,6 +226,16 @@ def dispatch(
             repo.package()
             repo.upload(url=lightning_app_release.source_upload_url)
 
+            # check if user has sufficient credits to run an app
+            # if so set the desired state to running otherwise, create the app in stopped state,
+            # and open the admin ui to add credits and running the app.
+            has_sufficient_credits = self._project_has_sufficient_credits(project, app=self.app)
+            app_release_desired_state = (
+                V1LightningappInstanceState.RUNNING if has_sufficient_credits else V1LightningappInstanceState.STOPPED
+            )
+            if not has_sufficient_credits:
```

Comment:
I am not sure about that, especially since `has_sufficient_credits` only indicates if user has `> 1` credit. Someone who has less than 1 credit can still run a Flow only app for few hours I believe

Reply:
this is coming from product, users can still run the app, they will see the warning as in the description.

re `has_sufficient_credits` only checking >1, this is a stopgap solution.